### DISCUSSION TITLE
Does Clair use dockershim internally?

### DISCUSSION THREAD

**Discussion by subramaniank08** [quay/clair#1776] [General]
With dockershim being deprecated, we are evaluating all the open source projects that we are using for docketshim's dependency. Does Clair internally use dockershim or any other docker api to fetch, index and scan layers?

We are planning to use containerd instead of dockershim as the runtime. Hope this wont affect Clair's static analysis of vulnerabilities in containers.

**Comment by hdonnay** [comment]
`Clair` doesn't use any docker packages. `Clairctl` uses some docker packages indirectly via `go-containerregistry`.

**Reply by subramaniank08** [reply]
Does it mean we will have to do something while switching to containerd? Any documentation on this?

**Reply by hdonnay** [reply]
No; `clair` does not use any container runtime.
